### ISSUE TITLE
Message#from_yaml results in invalid parts when body is not present

### ISSUE THREAD

**Issue by angelcarballo** [mikel/mail#1038]
Found this behaviour while testing serialization and deserialization of messages via YAML.

The following spec fails (at least on `master` and `2.6.4`)

``` ruby
context 'when body is not present' do
  it 'should encode after deserializing' do
    message = Mail::Message.new(
      to: 'someone@somewhere.com',
      cc: 'someoneelse@somewhere.com',
      bcc: 'someonesecret@somewhere.com',
      subject: 'subject'
    )
    serialized = message.to_yaml
    deserialized = Mail::Message.from_yaml(serialized)
    expect{ deserialized.encoded }.not_to raise_error
  end
end
```

The `#from_yaml` method sets `@body` directly, which after YAML serialization will contain an Array on `parts` instead of a `PartList` instance.

Is this a bug, or it's the expected behaviour (making the body mandatory) ?

**Comment by jeremy** [comment]
```ruby
>> Mail::Message.from_yaml(Mail.new.to_yaml).encoded
ArgumentError: wrong number of arguments (given 1, expected 0)
	from /Users/jeremy/source/mail/lib/mail/message.rb:1801:in `sort!'
```
ಠ_ಠ

**Comment by jeremy** [comment]
Caused by #782

**Comment by dragosmiron** [comment]
still failing on 2.7.0 @jeremy

**Comment by jeremy** [comment]
@dragosmiron Expected; thanks for the report. Pull requests welcome 😊

**Comment by dragosmiron** [comment]
But the fix is above in the PR, and in master, just not in a release yet. Any idea when the next release might be? We had to fork the gem to benefit from your 'master' code

**Comment by jeremy** [comment]
@dragosmiron Got a link to the fix you're referencing?

**Comment by dragosmiron** [comment]
https://github.com/mikel/mail/pull/1148

**Comment by jeremy** [comment]
> But the fix is above in the PR, and in master, just not in a release yet.

#1148 was not merged to master. The PR is a workaround for a symptom of the problem, not a fix for the underlying issue, so it was closed as a duplicate of this issue.

**Comment by dragosmiron** [comment]
cool, https://github.com/mikel/mail/pull/1148 is good enough until this is somehow fixed, so we had to fork the gem.

**Comment by dragosmiron** [comment]
hello, two years later, has this issue or the one from https://github.com/mikel/mail/pull/1148 been fixed?

**Comment by dragosmiron** [comment]
also, why don't we merge #1148 ? we've been running for 2 years on a fork of that with no more issues because of this error. has development stalled?

**Comment by jeremy** [comment]
@dragosmiron Please do implement a fix for the root cause and push a PR! Great chance to "scratch your own itch" on this project. (Again, #1148 conceals the bug rather than fixing it.)

**Comment by chrisradford** [comment]
I'm unable to reproduce this bug on the latest master, running both the spec in the description, the line [here](https://github.com/mikel/mail/issues/1038#issuecomment-255165240) and the spec from #1148 (which tested the "fix" in that PR), they are all passing for me.

@dragosmiron are you able to see if you can still reproduce this as it looks like it may have been accidentally fixed but I can't find where or when.

**Comment by jeremy** [comment]
Addressed by #1347. Thanks @lesleh!

**Comment by BuonOmo** [comment]
@jeremy this issue is closed, however, there never has been a new patch or minor including it. When should we expect that ?
